port/unix/omrsock.c

```diff
@@ -76,8 +76,40 @@ omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_add
 
 int32_t
 omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *service, omrsock_addrinfo_t hints, omrsock_addrinfo_t result)
-{
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+{	
+	omr_os_addrinfo *addrInfoResults = NULL;
+	omr_os_addrinfo *addrInfoHints = NULL;
+	int32_t count = 0;
+	int32_t rc = 0;
+	
+	if (NULL == result) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	
+	if (NULL == hints) {
+		rc = portLibrary->sock_getaddrinfo_create_hints(portLibrary, &hints, 0, 0, 0, 0);
```

Comment:
who will free (remove) the hints?

Reply:
This is incorrect behaviour, the right thing to do is:
```c++
omr_os_addrinfo* os_hints = NULL;
if (NULL != hints) {
  os_hints = hints->addrInfo;
}
```

[original impl](https://github.com/eclipse/openj9/blob/master/runtime/port/unix/j9sock.c#L856-L858)

Reply:
@rwy0717 @babsingh Actually, I used `omr_os_addrinfo *addrInfoHints = NULL;` instead of `os_hints` but the functionality, I think should be the same because in the next line is 
```addrInfoHints = (omr_os_addrinfo *)hints->addrInfo;```.
I was thinking that if hints is NULL, then we will create a hints for them using all 0s which means just no requirement on family, socket type, protocol or flags. This means like if it doesn't fail, then hints after this will no longer be NULL and an invalid argument. If it does fail, we can return the error from the function call (in omrsock_getaddrinfo_create_hints).

The hints allocation is using the per thread buffer so I think it'll replace current values in the hints buffer with all 0s. I was thinking after we set the &hints (using omrsock_getaddrinfo_create_hints) so that hints point to the per thread buffer with all 0s, we can use the new hints to pass into the getaddrinfo system function later on.